service/src/main/java/com/ge/predix/acs/obligation/management/ObligationServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ObligationServiceImpl.class);
+
+    @Autowired
+    private ObligationRepository obligationRepository;
+
+    @Autowired
+    private ZoneResolver zoneResolver;
+
+    private final JsonUtils jsonUtils = new JsonUtils();
+
+    @Transactional
+    @Override
+    public void upsertObligation(@Valid final Obligation obligation) {
+        try {
+            ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+            upsertObligation(zone, obligation);
+        } catch (Exception e) {
+            handleException(e, obligation.getName());
+        }
+    }
+
+    @Transactional
+    @Override
+    public void upsertObligations(final List<Obligation> obligations) {
+        ValidList<Obligation> validObligations = new ValidList<Obligation>(obligations);
+        upsertObligations(validObligations);
+    }
+
+    private void upsertObligations(@Valid final ValidList<Obligation> obligations) {
+        String obligationName = "";
+        try {
+            ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+            for (Obligation obligation : obligations) {
+                obligationName = obligation.getName();
+                upsertObligation(zone, obligation);
+            }
+        } catch (Exception e) {
+            handleException(e, obligationName);
+        }
+    }
+
+    @Override
+    public List<Obligation> getObligations() {
+        ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+        ArrayList<Obligation> result = new ArrayList<>();
+        List<ObligationEntity> obligationEnityList = this.obligationRepository.findByZone(zone);
+        for (ObligationEntity obligationEntity : obligationEnityList) {
+            result.add(this.jsonUtils.deserialize(obligationEntity.getObligationJson(), Obligation.class));
+        }
+        return result;
+    }
+
+    @Override
+    public Obligation getObligation(final String obligationName) {
+        ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+        ObligationEntity obligationEntity = this.obligationRepository.getByZoneAndObligationId(zone, obligationName);
+        if (obligationEntity != null) {
+            return this.jsonUtils.deserialize(obligationEntity.getObligationJson(), Obligation.class);
+        }
+        LOGGER.debug("No policy set found for obligationName = {},  zone = {}.", obligationName, zone);
+        return null;
+    }
+
+    @Override
+    public Boolean deleteObligation(final String obligationName) {
+        ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+        ObligationEntity obligationEntity = this.obligationRepository.getByZoneAndObligationId(zone, obligationName);
+        if (obligationEntity != null) {
+            LOGGER.info("Found an existing obligation obligationName={}, zone={}, deleting now.", obligationName,
+                    zone.getName());
+            this.obligationRepository.delete(obligationEntity);
```

Comment:
Here we also need to make sure that an obligation is not being used in policies, if an obligation is being used in policy we cannot delete it.